pilot/cmd/pilot-agent/status/server_test.go

```diff
@@ -183,10 +185,16 @@ func TestHttpsAppProbe(t *testing.T) {
 	go server.Run(context.Background())
 
 	var statusPort uint16
-	for statusPort == 0 {
+	if err := retry.UntilSuccess(func() error {
```

Comment:
Can't you just add a channel into `Server` that is closed when the `statusPort` is set?
Would that be overkill?
```
type Server struct {
	...
	statusPort          uint16
	statusPortSet       chan struct{}
	...
}
```

```
func (s *Server) Run(ctx context.Context) {
	...
		s.mutex.Lock()
		s.statusPort = uint16(allocatedPort)
		s.mutex.Unlock()
		close(s.statusPortSet)
	...
}
```

And here:
```
	select {
	case <-server.statusPortSet:
	case <-time.After(30*time.Second):
		t.Fatal("failed to getport after 30 seconds")
	}
	...
```

Reply:
That is probably a safer fix but more invasive to the application code just for the sake of testing. I'm open to either solution though